### PR TITLE
Getting Started Update

### PR DESCRIPTION
**Pull request by ardasener**
This PR updates the getting started to match the current version of the library. Specifically,
- The code snippets are modified to promote the Base classes
- Some advice regarding shortening of namespaces and template arguments is added.
- Compilation steps are updated.
- Some typos from previous commits are fixed.

### CODE REVIEW

**Comment by AmroAlJundi** [comment]
I can't link to lines that werent' changed, but in `docs/pages/getting_started/installation_building.md:18` we actually require CMake 3.12 without CUDA and 3.18 with CUDA.

**Comment by AmroAlJundi** [comment]
In `docs/pages/getting_started/usage.md:11` "the following ~the~ command..."

File: docs/pages/getting_started/usage.md [hunk 1]
```diff
@@ -78,14 +80,52 @@ This can be useful to reduce compile times if the header only build is being use
...
+using sbfe = spasebase::feature;
+using sbob = sparsebase::object;
+using sbpe = sparsebase::preprocess;
+using sbco = sparsebase::context;
+using sbut = sparsebase::utils;
+```
+
+## Template Types
+
+To be flexible, efficient and safe, SparseBase classes take most of the types they use as
```
Comment (AmroAlJundi) [inline docs/pages/getting_started/usage.md:104]: I like this. Gonna use it in presentations from now on.

**Comment by AmroAlJundi** [comment]
In docs/pages/getting_started/usage.md:140 "also contains" -> "which contains"

**Comment by AmroAlJundi** [comment]
In `docs/pages/getting_started/usage.md:267` I think this should be inside a note or a warning.

**Comment by AmroAlJundi** [comment]
What do you think about deprecating the "Working with graphs" section of usage? We really have no functionalities for these guys and this might give users the idea that we have graph-specific stuff. I know you have the note at the end but like, what do you think?

Either way, can you change "SparseFormat" to "Format" in that section?

**Comment by AmroAlJundi** [comment]
In the "Ordering" section at the end of usage, can you mention the parameters part? That each ordering has a parameters struct that you need to pass an instance of, or use bracket initialization for? I have similar text written [the "Result" section of this how-to guide](https://github.com/sparcityeu/sparsebase/pull/206/files#diff-f1535e646e160420c19f92ba7b34a5c4bdee0f721be6462850839af22fba9287).

**Review by AmroAlJundi — CHANGES_REQUESTED** [review]
Besides a few changes I requested as conversion, looks great.

File: docs/pages/getting_started/usage.md [hunk 2]
```diff
@@ -209,33 +283,53 @@ auto g = sparsebase::object::Graph<vertex_type, edge_type, value_type>(data);
...
 
-Various orderings can be generated for a graph using the ``ReorderPreprocessType`` classes. 
-Currently these include ``RCMReoder`` and ``DegreeReorder``. There is also a ``GenericReorder`` class
-allowing the users to define their own custom orderings. For more details, please see the examples
-in the Github repository.
+Sparse data formats can be reordered easily using the `ReorderBase` class.
+
+```cpp
+sparsebase::context::CPUContext cpu_context;
+IDType* order = ReorderBase::Reorder<DegreeReorder>(format, {&cpu_context}, true);
```
Comment (AmroAlJundi) [inline docs/pages/getting_started/usage.md:298]: Actually, `DegreeReorder` requires a `DegreeReorderParams` containing an `ascending` boolean.

File: docs/pages/getting_started/usage.md [hunk 3]
```diff
@@ -209,33 +283,53 @@ auto g = sparsebase::object::Graph<vertex_type, edge_type, value_type>(data);
...
+
+
+Alternatively the user can directly call the underlying reordering classes.
+Below you can see an example of an RCM reordering of a graph using this method.
 
-Below you can see an example of an RCM reordering of a graph.
 ```cpp
-sparsebase::preprocess::RCMReorder<vertex_type, edge_type, value_type> orderer(1, 4);
-sparsebase::format::Format<vertex_type, edge_type, value_type> * con = g.get_connectivity();
+sparsebase::preprocess::RCMReorder<int,int,float> orderer(1, 4);
```
Comment (AmroAlJundi) [inline docs/pages/getting_started/usage.md:308]: Also, this is outdated. RCMReorder no longer takes any parameters.

File: docs/pages/getting_started/usage.md (hunk 2 above)
Comment (ardasener) [reply docs/pages/getting_started/usage.md:298]: Hmmm. I thought they were optional. I see now that they are not. I will change it but not sure if I like it.

**Review by AmroAlJundi — DISMISSED** [review]

**Review by AmroAlJundi — APPROVED** [review]

**Pull request merged**